dialogue-core/src/main/java/com/palantir/dialogue/core/PinUntilErrorNodeSelectionStrategyChannel.java

```diff
@@ -133,7 +133,10 @@ LimitedChannel getCurrentChannel() {
         int pin = currentPin.get();
         PinChannel channel = nodeList.get(pin);
 
-        Optional<ListenableFuture<Response>> maybeResponse = channel.maybeExecute(endpoint, request, limitEnforcement);
+        // n.b. StickyAttachments.maybeExecute uses the delegate PinChannel, which bypasses the FutureCallback
```

Comment:
Why not include this instrumentation?

Reply:
Granted we don't use this strategy with sticky sessions, it's still a bit sucky not to handle it correctly.

Reply:
I think if PinChannel knows it's current index, as well as the stable index, you won't have to allocate; then just need to change the recovery logic

```
initialShuffle.indexOf(limitedChannel)
```

To actually only compare the underlying LimitedChannels?

Reply:
Discussed: this is fine because the initial request will still be tracked, so if that happens to fail, we'll move hosts anyway. Not worth being complete. @carterkozak just to add a comment why this is fine